MyLibrary/BusinessLogicTests/AuthorServiceTests.cs

```diff
@@ -0,0 +1,209 @@
+        /// Tests the method GetAuthorById
+        /// Correct input values
+        /// Returned correct author
+        /// </summary>
+        /// <param name="id">The author identifier</param>
+        /// <param name="name">The author name</param>
+        [TestMethod]
+        [DataRow(1u, "TestAuthor")]
+        public void GetAuthorById_CorrectId_ReturnedCorrectAuthor(uint id, string name)
+        {
+            // Arrange
+            Author expectedAuthor = new Author(name);
+            var mockDataProvider = new Mock<IDataProvider>();
+            mockDataProvider.SetupGet(x => x.GetAuthors).Returns(new List<Author> { expectedAuthor });
+            var libraryService = new AuthorService(mockDataProvider.Object);
+
+            // Act
+            Author actualAuthor = libraryService.GetById(id);
+
+            // Assert
+            Assert.AreEqual(expectedAuthor.FullName, actualAuthor.FullName);
+        }
+
+        /// <summary>
+        /// Tests the method GetAuthorById
+        /// Incorrect input value for id of author
+        /// Returned null
+        /// </summary>
+        [TestMethod]
+        public void GetAuthorById_IncorrectId_ReturnedNull()
+        {
+            // Arrange
+            Author author = new Author("TestAuthor");
+            var mockDataProvider = new Mock<IDataProvider>();
+            mockDataProvider.SetupGet(x => x.GetAuthors).Returns(new List<Author> { author });
+            var libraryService = new AuthorService(mockDataProvider.Object);
+            uint id = 100;
+
+            // Act
+            Author actualAuthor = libraryService.GetById(id);
+
+            // Assert
+            Assert.IsNull(actualAuthor);
+        }
+
+        /// <summary>
+        /// Tests the method RemoveAuthor
+        /// Correct input values
+        /// Returned identifier of removed  author
+        /// </summary>
+        /// <param name="id">The identifier of author</param>
+        /// <param name="name">The name of author</param>
+        [TestMethod]
+        [DataRow(1u, "TestAuthor")]
+        public void RemoveAuthor_CorrectInput_ReturnedIdOfRemovedAuthorAndDeleteAllBookAuthorPairs
+            (uint id, string name)
+        {
+            // Arrange
+            uint? expectedId = id;
+            Author author = new Author(name);
+            List<BookAuthorPair> pairBookAuthor = new List<BookAuthorPair>()
+            {
+                new BookAuthorPair(1, 2),
+                new BookAuthorPair(2, 3)
+            };
+            var mockDataProvider = new Mock<IDataProvider>();
+            mockDataProvider.SetupGet(x => x.GetAuthors).Returns(new List<Author> { author });
+            mockDataProvider.SetupGet(p => p.GetPairBookAuthor).Returns(pairBookAuthor);
+            var libraryService = new AuthorService(mockDataProvider.Object);
+
+            // Act
+            uint? actualId = libraryService.Remove(id);
+            Author actualAuthor = libraryService.GetById(id);
```

Comment:
I have just noticed. 
Commonly try to use only 1 method from a tested class in one unit test. There are possibilities to check the result of method work via mock collection. It should have all data with all updates. As you are controlling it. 
Just FYI. I'm not asking to rewrite tests.
This is an issue because if someone will break GetById method. It will cause fails of remove author tests. As you are depended on it here etc.

Reply:
Understand. In my 4th level I tried to awoid this situation